Update labels to use kubeless key instead of app

rsmitty opened the pull request:
I was getting started checking on out kubeless on my minikube install and noticed that none of the `kubeless topic` commands were working for me. I was able to determine that this is because the topic commands search for a label of `kubeless=kafka` to get the pods it issues commands against. That was [introduced here](https://github.com/kubeless/kubeless/commit/7f74dba0680a9a2698b8a71860ab7d322d104dd5).  

The current manifest for kafka, however, still uses the `app=kafka` label. This PR updates the labels for the kafka stateful set to match the expected `kubeless=kafka`.

sebgoa commented:
@rsmitty hey :) go to see you here.

How did you start kubeless ? Did you use the manifest that is in the release ?

rsmitty commented:
Yep, I got started with the normal installation instructions. Created the namespace and then issued the curl -> jq pipe that points to the latest release yaml (kubeless-0.0.16.yaml). That yaml contains the `app: kafka` label for the kafka StatefulSet that was causing me issues.

rsmitty commented:
Here's a dump of the error that the kubeless binary throws when that label is present:

```
$ kubeless topic list
panic: runtime error: index out of range

goroutine 1 [running]:
main.execCommand(0xc42018b590, 0x5, 0x5, 0x293c78b, 0x8)
	/home/travis/gopath/src/github.com/kubeless/kubeless/cmd/kubeless/topicCreate.go:60 +0x584
main.glob..func11(0x36a6840, 0x36d3a68, 0x0, 0x0)
	/home/travis/gopath/src/github.com/kubeless/kubeless/cmd/kubeless/topicList.go:35 +0x19b
github.com/kubeless/kubeless/vendor/github.com/spf13/cobra.(*Command).execute(0x36a6840, 0x36d3a68, 0x0, 0x0, 0x36a6840, 0x36d3a68)
	/home/travis/gopath/src/github.com/kubeless/kubeless/vendor/github.com/spf13/cobra/command.go:603 +0x22b
github.com/kubeless/kubeless/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4201cf8c0, 0x36a5b80, 0x36a61e0, 0x36a6ea0)
	/home/travis/gopath/src/github.com/kubeless/kubeless/vendor/github.com/spf13/cobra/command.go:689 +0x339
github.com/kubeless/kubeless/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4201cf8c0, 0xc420000340, 0xc420000340)
	/home/travis/gopath/src/github.com/kubeless/kubeless/vendor/github.com/spf13/cobra/command.go:648 +0x2b
main.main()
	/home/travis/gopath/src/github.com/kubeless/kubeless/cmd/kubeless/kubeless.go:41 +0x2f
```

sebgoa commented:
ok so the manifest that is in the release is generated from a ksonnet file: https://github.com/kubeless/kubeless/blob/master/kubeless.jsonnet

that's where we need to fix the labels.

good catch

rsmitty commented:
Ah, cool. I'll update my PR to update it there instead.

sebgoa commented:
and you can check the generated manifest with jsonnet...(bit of a rabbit hole, but we are eating our own dog food)

rsmitty commented:
Alright, updated the jsonnet and output the yaml with kubecfg (via the `make kubeless.yaml` command). I deleted the kubeless deployment with the uninstall script and redeployed with `kubectl create -f kubeless.yaml`. Seems to be working as expected!

```
$ kubeless topic list

$ kubeless topic create test
Created topic "test".

$ kubeless topic list
test
```

ngtuna commented:
Ah... that's a missed bit once I turned the kubeless installation into using `kubeless.jsonnet` file. Deployment manifests in `/manifest` folder still using `app=kafka/app=zookeeper` label. Thanks for the catch @rsmitty !!! 

Would you mind send a follow-up commit for updating `kubeless=zookeeper` label ? That hasn't been used but just in case of keeping consistency of labels.

rsmitty commented:
@ngtuna Think I got 'em all now. Thanks!

ngtuna commented:
Thanks @rsmitty for the patch 🥇. LGTM. Merging
